ui/src/alerting/components/endpoints/EndpointOverlayFooter.tsx

```diff
@@ -12,39 +12,71 @@ import {
 // Hooks
 import {useEndpointState} from './EndpointOverlayProvider'
 
+// API
+import {testExistingEndpoint} from 'src/alerting/actions/notifications/endpoints'
+
 // Types
 import {NotificationEndpoint, RemoteDataState} from 'src/types'
 
-interface Props {
+export interface Message {
+  text: string
+  remoteState: RemoteDataState
+}
+interface OwnProps {
   saveButtonText: string
-  onSave: (endpoint: NotificationEndpoint) => void
+  onSave: (endpoint: NotificationEndpoint) => Promise<void>
+  onTest: (endpoint: NotificationEndpoint) => Promise<void>
```

Comment:
Is there a reason why we've passed `onTest` all the way down here without using it?